fetcher/fetcher.go

```diff
@@ -61,24 +66,34 @@ func NewPool(limit, timeout int, repos []string) Pool {
 	return f
 }
 
-func (p *Pool) Fetch(url string) (io.ReadCloser, error) {
-	client := &http.Client{
-		Timeout: time.Duration(p.timeout) * time.Second,
-	}
-	resp, err := client.Get(url)
-	if err != nil {
-		return nil, err
-	}
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to fetch with: %d", resp.StatusCode)
+func (p *Pool) RetryFetch(url string) (io.ReadCloser, error) {
```

Comment:
Would there be a case where Fetching does not use retry functionality?

Would this function ever be called outside of the Pool?

Reply:
I don't see why it would be.

Reply:
Should `RetryFetch` be unexported?

Reply:
Yeah, that's a good point, it should.

Reply:
Same for `TryRepos` and `TryRepo` I guess.